pre_commit/languages/docker.py

```diff
@@ -76,7 +78,28 @@ def install_environment(
         os.mkdir(directory)
 
 
+def _docker_is_rootless() -> bool:  # pragma: win32 no cover
+    returncode, stdout, stderr = cmd_output(
+        'docker', 'system', 'info',
+    )
```

Comment:
It would be even faster and simplify both the logic below and the mock output used in testing to only query Docker/Podman for the field we care about. From [the rootless-docker GitHub Action's README](https://github.com/ScribeMD/rootless-docker#rootless-docker), `docker info --format "{{ .ClientInfo.Context }}"` outputs `rootless` in Docker when in rootless mode, and other strings (most commonly `default`, I believe) otherwise. I suspect, based on [Podman's documentation](https://docs.podman.io/en/latest/markdown/podman-info.1.html#example), that `podman info --format "{{ .host.security.rootless }}"` outputs `true` when in rootless mode and `false` otherwise. [Go Templates](https://pkg.go.dev/text/template#hdr-Actions) are quite powerful, so we can say: `docker info --format '{{ if .ClientInfo.Context }} {{ eq .ClientInfo.Context "rootless" }} {{ else }} {{ .host.security.rootless }} {{ end }}'`. This command outputs `true` when running in rootless mode and `false` otherwise on both Docker and Podman, although I have only tested it on Docker. I don't have Podman installed, so hopefully someone who does can straighten me out if I'm mistaken. Disclosure: I am the author of the rootless-docker GitHub Action.

Reply:
Here on my machine (Linux Mint 19.3, Docker version 20.10.16, build aa7e414) your suggestion returns `default` for rootless docker, as `rootless` seems to be part of the `Security Options` under `Server`.

Reply:
Thanks for letting me know. Bummer that the behavior isn't consistent across our machines even on the exact same build of Docker. I am on Ubuntu 22.04, but I wonder whether the difference has more to do with how we installed rootless Docker? It's possible to detect `rootless` mode under `SecurityOptions` too (e.g., `docker info --format '{{ range .SecurityOptions }}{{ if eq . "name=rootless" }}true{{ end }}{{ end }}'`), FWIW.

Reply:
IIRC I followed the [official instructions](https://docs.docker.com/engine/security/rootless) for Ubuntu. Maybe you chose to set the client context explicitly where I went for the `DOCKER_HOST` environment variable? Both options are listed in the instructions: https://docs.docker.com/engine/security/rootless/#client

Reply:
Yes, you are correct. I suppose one could check the `DOCKER_HOST` environment variable before running `docker info`.